AssemblerLib/MeshComponentMap.cpp

```diff
@@ -259,4 +263,34 @@ std::vector<GlobalIndexType> MeshComponentMap::getGlobalIndicesByComponent(
     return global_indices;
 }
 
+std::vector<GlobalIndexType>
+MeshComponentMap::getIndicesForComponent(
+        const std::vector<GlobalIndexType>& cnt,
+        const unsigned component_id) const
+{
+    std::vector<GlobalIndexType> new_cnt;
+    auto const N = cnt.size() / _num_components;
```

Comment:
That's a valid point. So far I've only considered scalar components which all operate on basically the same mesh, i.e. all have same order finite elements.
I will think about that.